### PR TITLE
Run promises in order

### PR DESCRIPTION
**Pull request by hardillb**
## Description

possible for creating/deleting k8s artefacts to fail if done out of order or too quickly.

## Related Issue(s)

#114

## Checklist

 - [ ] I have read the [contribution guidelines](https://github.com/flowforge/flowforge/blob/main/CONTRIBUTING.md)
 - [ ] Suitable unit/system level tests have been added and they pass 
 - [ ] Documentation has been updated
    - [ ] Upgrade instructions
    - [ ] Configuration details
    - [ ] Concepts
 - [ ] Changes `flowforge.yml`?
    - [ ] Issue/PR raised on `flowforge/helm` to update ConfigMap Template
    - [ ] Issue/PR raised on `flowforge/CloudProject` to update values for Staging/Production

## Labels

 - [ ] Backport needed? -> add the `backport` label
 - [ ] Includes a DB migration? -> add the `area:migration` label

### CODE REVIEW

**Comment by elenaviter** [comment]
Hello @hardillb,
I've opened followup [PR](https://github.com/FlowFuse/driver-k8s/pull/118) on this same branch.
The code was staged on our systems and shown the problems it address are solved:
- suspend/start project is not guaranteed to succeed
- projects auth token deteriorates after FF upgrade 

We are aiming to integrate it at the earliest convenience. 
Thank you in advance for considering our request! We are looking forward to your valuable feedback.

**Review by elenaviter — APPROVED** [review]

**Comment by hardillb** [comment]
I've tested this locally on my K8s rig and it looks good so far.

**Comment by hardillb** [comment]
@ppawlowski when you get moment can you review this so I can deploy it to staging for more testing. Thanks

File: kubernetes.js [hunk 1]
```diff
@@ -372,63 +373,116 @@ const createProject = async (project, options) => {
...
+                await this._k8sAppApi.readNamespacedDeployment(project.safeName, this._namespace)
+                clearInterval(pollInterval)
+                resolve()
+            } catch (err) {
+                // hmm
+                counter++
+                if (counter > this._k8sRetries) {
+                    this._app.log.error(`[k8s] Project ${project.id} - timeout waiting for Deployment`)
+                    reject(new Error('Timed out to creating Deployment'))
+                }
```
Comment (knolleary) [inline kubernetes.js:413]:
```suggestion
                if (counter > this._k8sRetries) {
                    clearInterval(pollInterval)
                    this._app.log.error(`[k8s] Project ${project.id} - timeout waiting for Deployment`)
                    reject(new Error('Timed out to creating Deployment'))
                }
```

**Review by knolleary — CHANGES_REQUESTED** [review]
Spotted some intervals that wouldn't be stopped in case of an error.

**Review by knolleary — APPROVED** [review]

**Pull request merged**